ts/main.ts

```diff
@@ -118,4 +153,209 @@ function renderCard(data: Movie): HTMLElement {
   return $outerColumn;
 }
 
+function renderMovieDetails(data: Movie): void {
+  const $mainWrapper = document.createElement('div');
+  $mainWrapper.setAttribute('class', 'main-wrapper row flex-wrap');
+
```

Comment:
Instead of creating new elements every time you want to view details of movies, you should create the elements in your HTML, query for them at the top of your `main.ts` file, and then update the attribute values in your movieDetails function. It's more work to have to create and delete elements from the DOM versus updating the content of the elements.

I also didn't notice a part where you are removing the added elements so would there be multiple details if you went back and forth without refreshing the page?